codebuild/README.md

```diff
@@ -0,0 +1,926 @@
+
+```hcl
+module "github_runner" {
+  source = "github.com/Enterprise-CMCS/terraform-aws-codebuild-github-runner"
+
+  # GitHub App authentication (module creates the connection!)
+  auth_method            = "github_app"
+  github_connection_name = "github-codebuild-runners"
+
+  # Repository configuration
+  github_owner      = "your-org"
+  github_repository = "your-repo"
+  project_name      = "my-project"
+  environment       = "prod"
+
+  tags = {
+    Team = "DevOps"
+  }
+}
+
+# Module outputs authorization instructions
+output "setup_instructions" {
+  value = module.github_runner.setup_complete
+}
+```
+
+**Step 3: Authorize Connection** (One-time manual step)
+
+After `terraform apply`, the module output will show:
+
+1. Direct link to AWS Console
+2. Connection name and status
+3. Step-by-step authorization instructions
+
+Simply:
+
+1. Click the console link
+2. Find your connection (status: PENDING)
+3. Click "Update pending connection"
+4. Authorize with GitHub and select your GitHub App
+5. Done! Connection status → AVAILABLE
+
+#### Alternative: Use Existing Connection
+
+If you already have an authorized CodeConnections connection:
+
+```hcl
+module "github_runner" {
+  source = "github.com/Enterprise-CMCS/terraform-aws-codebuild-github-runner"
+
+  # Use existing connection
+  auth_method           = "github_app"
+  github_connection_arn = "arn:aws:codeconnections:us-east-1:123456789:connection/abc-123"
+
+  github_owner      = "your-org"
+  github_repository = "your-repo"
+  project_name      = "my-project"
+  environment       = "prod"
+}
+```
+
+
+
+### Option B: Personal Access Token (PAT) Authentication
+
+### Step 0: Get AWS credential from Kion/Cloudtamer
+
+Get temporary AWS credential from Kion/Cloudtamer and run in on your terminal
+
+### Step 1: Generate GitHub Token
+
+Create new or update existing PAT with the following scopes:
+
+- `repo` - Full control of repositories
+- `admin:repo_hook` - Webhook management
+- `admin:org` - Organization administration (for organization repositories)
+
+### Step 2: Create AWS Secret
+
+```bash
+# Create secret with placeholder value first
+aws secretsmanager create-secret \
+  --name "github/actions/runner-token" \
+  --secret-string "placeholder"
+
+# Update with actual token value (keep token secure, don't store in shell history)
+aws secretsmanager put-secret-value \
+  --secret-id "github/actions/runner-token" \
+  --secret-string "ghp_your_actual_token_here"
+```
+
+
+### Step 3: update terraform.tfvars
+
+cd test-module
+cp terraform.tfvars.example terraform.tfvars
+
+Update owner, repo, project name, environment and tags
+
+
+### Step 4: Deploy
+
+```bash
+terraform init -upgrade
+terraform plan
+terraform apply
+```
+
+### Step 5: Test self-hosted runner with a wokflow file
+
+Create a test github workflow file to test your self hosted runner. You can use the sample test-new-runner.yml file in the test-module directory.
```

Comment:
Step 5 instructions would render correctly and be easier to read if workflow references like `test-new-runner.yml`, `runs-on:`, `codebuild-demo-runner-dev-runner-${github.run_id}-${github.run_attempt}` were surrounded with backticks for inline code formatting.